source/common/network/lc_trie.cc

```diff
@@ -9,26 +9,26 @@ const uint32_t LcTrie::LcTrieInternal<IpType, address_size>::MAXIMUM_CIDR_RANGE_
 
 LcTrie::LcTrie(const std::vector<std::pair<std::string, std::vector<Address::CidrRange>>>& tag_data,
                double fill_factor, uint32_t root_branching_factor) {
-  std::vector<IpPrefix<Ipv4>> ipv4_prefixes;
-  std::vector<IpPrefix<Ipv6>> ipv6_prefixes;
+  BinaryTrie<Ipv4> ipv4_temp;
```

Comment:
I think it would be helpful for the reader for there to be more comments here. If it was me describing the optimization, I would add some ASCII art of what the trees look like to help as well.